lumigator/backend/backend/tests/conftest.py

```diff
@@ -262,32 +264,46 @@ def fake_s3fs() -> S3FileSystem:
 @pytest.fixture(scope="function")
 def fake_s3_client(fake_s3fs) -> S3Client:
     """Provide a fake S3 client using MemoryFileSystem as underlying storage."""
-    os.environ["AWS_ACCESS_KEY_ID"] = os.environ.get("AWS_ACCESS_KEY_ID", "lumigator")
-    os.environ["AWS_DEFAULT_REGION"] = os.environ.get("AWS_DEFAULT_REGION", "us-east-2")
-    os.environ["AWS_SECRET_ACCESS_KEY"] = os.environ.get(
-        "AWS_SECRET_ACCESS_KEY", "lumigator"
-    )  # pragma: allowlist secret
-    os.environ["AWS_ENDPOINT_URL"] = os.environ.get("AWS_ENDPOINT_URL", "http://localhost:9000")
     return FakeS3Client(MemoryFileSystem.store)
 
 
 @pytest.fixture(scope="function")
 def boto_s3_client() -> S3Client:
     """Provide a real S3 client."""
-    os.environ["AWS_ACCESS_KEY_ID"] = os.environ.get("AWS_ACCESS_KEY_ID", "lumigator")
-    os.environ["AWS_DEFAULT_REGION"] = os.environ.get("AWS_DEFAULT_REGION", "us-east-2")
-    os.environ["AWS_SECRET_ACCESS_KEY"] = os.environ.get(
-        "AWS_SECRET_ACCESS_KEY", "lumigator"
-    )  # pragma: allowlist secret
-    os.environ["AWS_ENDPOINT_URL"] = os.environ.get("AWS_ENDPOINT_URL", "http://localhost:9000")
-    return boto3.client("s3")
+    aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID", "lumigator")
+    aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY", "lumigator")
+    aws_endpoint_url = os.environ.get("AWS_ENDPOINT_URL", "http://localhost:9000")
+    aws_default_region = os.environ.get("AWS_DEFAULT_REGION", "us-east-2")
+
+    return boto3.client(
+        "s3",
+        endpoint_url=aws_endpoint_url,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+        region_name=aws_default_region,
+    )
 
 
 @pytest.fixture(scope="function")
-def boto_s3fs() -> S3FileSystem:
-    """Provide a real s3fs client."""
-    s3fs = S3FileSystem()
-    mock_s3fs = MagicMock(wraps=s3fs)
+def boto_s3fs() -> Generator[S3FileSystem, None, None]:
+    """Provide a real s3fs client wrapped with a mock to intercept calls."""
+    aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID", "lumigator")
+    aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY", "lumigator")
+    aws_endpoint_url = os.environ.get("AWS_ENDPOINT_URL", "http://localhost:9000")
+    aws_default_region = os.environ.get("AWS_DEFAULT_REGION", "us-east-2")
+
+    s3fs = S3FileSystem(
+        key=aws_access_key_id,
+        secret=aws_secret_access_key,
+        endpoint_url=aws_endpoint_url,
+        client_kwargs={"region_name": aws_default_region},
+    )
+
+    mock_s3fs = MagicMock(wraps=s3fs, spec=S3FileSystem)
+    mock_storage_options = MagicMock()
+    mock_storage_options.__getitem__.side_effect = lambda key: aws_endpoint_url if key == "endpoint_url" else None
```

Comment:
Again, is this really needed? Do storage options need to be specified?

Reply:
I've updated this based on your more succinct version. 👍🏼